docs/langchain.md

```diff
@@ -0,0 +1,96 @@
+# Deploying LangChain Agents on Fixie
+
+## Overview
+
+[LangChain](https://python.langchain.com/) is a popular framework for developing applications built on top of Large Language Models. Fixie supports LangChain as a first-class citizen. This means you can develop LangChain Agents/Chains/Prompts just as you typically would and then quickly deploy them onto Fixie infrastructure.
+
+## Why Deploy on Fixie?
+
+Deploying on Fixie gives you three major advantages:
+
+1. We host it for you (for free). Simply run `fixie deploy` and your Agent will be hosted in the cloud, complete with a REST API for calling it programatically and a shareable URL that will allow others to demo your agent in the browser.
+1. No need for an OpenAI key. Fixie will automatically take care of that, and supports 1,000 free requests per day, per user. So if your creation gets popular, you're not stuck footing the bill or asking people to mint their own key.
+1. It allows you to integrate with the rest of the Fixie ecosystem. This means other people in the community will be able to leverage your agent and vice-versa.
+
+## Getting Started
+
+First, you'll need to make sure you have `fixie` installed on your machine with `pip install fixieai`. This will install the Fixie CLI, which you'll need for deploying to Fixie. You'll also need a Fixie account and to authorize the Fixie CLI. You can do both by running `fixie auth`.
+
+Next, you'll need to tell Fixie what function to call to invoke your app. As an example, let's imagine something as simple as a [PromptTemplate](https://python.langchain.com/en/latest/getting_started/getting_started.html):
+
+```python
+from langchain.prompts import PromptTemplate
+from langchain.llms import OpenAI
+from langchain.chains import LLMChain
+
+llm = OpenAI(temperature=0.9)
+prompt = PromptTemplate(
+    input_variables=["product"],
+    template="What is a good name for a company that makes {product}?",
+)
+chain = LLMChain(llm=llm, prompt=prompt)
+chain.run("colorful socks")
+```
+
+In the above example, we're calling `chain.run()` directly inside of our code with a hardcoded value of "colorful socks". This isn't that useful, since it means we can only get company names for colorful socks! Instead, we want a user to be able to provide the variable themselves, and we want Fixie to pass that to us.
+
+To accomplish this, we're going to wrap `chain.run()` inside of a func. Here's our new version:
+
+```python
```

Comment:
You could do this as a `diff` instead, so it's more obvious what lines have changed:

```diff
from langchain.prompts import PromptTemplate
from langchain.llms import OpenAI
from langchain.chains import LLMChain

llm = OpenAI(temperature=0.9)
prompt = PromptTemplate(
  input_variables=["product"],
  template="What is a good name for a company that makes {product}?",
)

chain = LLMChain(llm=llm, prompt=prompt)

+def invoke_chain(input):
+    return chain.run(input)
```

But then ofc you lose the python highlighting, so maybe it's worse.

Reply:
Yeah, I'm not a big fan of diffs in these things. I prefer being able to just copy and paste to make things work.